lib/babe/epoch_handler.go

```diff
@@ -44,23 +46,34 @@ func newEpochHandler(epochNumber, firstSlot uint64, epochData *epochData, consta
 			epochData.threshold,
 			keypair,
 		)
-		if errors.Is(err, errOverPrimarySlotThreshold) {
+		if err == nil {
+			slotToProof[i] = proof
+			slotToIfPrimary[i] = true
+			logger.Debugf("epoch %d: claimed slot %d", epochNumber, i)
 			continue
-		} else if err != nil {
+		}
+		if !errors.Is(err, errOverPrimarySlotThreshold) {
+			return nil, fmt.Errorf("error running slot lottery at slot %d: %w", i, err)
+		}
+
+		proof, err = claimSecondarySlot(epochData.randomness, i, epochNumber, epochData.authorities, epochData.threshold, keypair)
+		if err != nil {
 			return nil, fmt.Errorf("error running slot lottery at slot %d: %w", i, err)
 		}
 
 		slotToProof[i] = proof
+		slotToIfPrimary[i] = false
 		logger.Debugf("epoch %d: claimed slot %d", epochNumber, i)
```

Comment:
duplicate of `runLottery` code - can you store primary and secondary proofs in separate maps potentially?